hibernate-core/src/main/java/org/hibernate/sql/ast/spi/SimpleFromClauseAccessImpl.java

```diff
@@ -52,7 +52,7 @@ public TableGroup findTableGroup(NavigablePath navigablePath) {
 
 	@Override
 	public TableGroup findTableGroupForGetOrCreate(NavigablePath navigablePath) {
-		final TableGroup tableGroup = findTableGroup( navigablePath );
+		final TableGroup tableGroup = locateTableGroup( navigablePath );
 		if ( parent != null && tableGroup instanceof VirtualTableGroup && tableGroup.getModelPart() instanceof EmbeddableValuedModelPart ) {
```

Comment:
```suggestion
		if ( parent != null && tableGroup != null ) {
```

Reply:
This would be nice, but not sure if the code below works fine with non embeddable paths like in this case. Ideally, we should change the code below though instead of adding this `locateTableGroup` method.